### DISCUSSION TITLE
Dynamic permission request on action click?

### DISCUSSION THREAD

**Discussion by nnx0r** [wxt-dev/wxt#437] [Q&A]
This is not WXT related, but I think it's worth asking this question as it may be helpful to the other ext. developers _(if possible at all)_.

My extension currently requires access to all sites and it bothers many users. But it's justified from my side, because the extension can work anywhere and content_script injection matches both `http://*/*` and `https://*/*` protocols - so access to all websites.

I was wondering, would it be possible to declare popup in a manifest, but then use the same browser action to toggle whether the popup will show up (if it has access to that site) or it will prompt browser to request access to that domain, before it can work. 

So if user approves, it would then add that domain to the Site access list and they can toggle it later (on/off) from the browser's interface. 

Not sure if that's possible, or I would need to declare `optional_host_permissions` for this? If so, that'd be impossible, because there are many video streaming platforms world-wide, so my `optional_host_permisions` would be huuuuge.

Please let me know, fellow extension developers. I do not have much knowledge in this area of permission requests.

**Comment by aklinker1** [comment]
I assume you're talking about Firefox. You could request `*://*/*` as an optional host permission, then use the chrome.scripting API to register a content script dynamically when you click the action icon or a button in your popup.

I don't know if that will prompt the user to allow the content script or not, unfortunately I bet it won't. But that's the only suggestion I have. 

The user might still have to enable each website manually on the extension settings page anyways...
